flat_pack.scad

```diff
@@ -0,0 +1,169 @@
+//openscad flat-pack joint library
+
+module finger_testing() {
+  difference() {
+    color("blue")
+      square([110, 40]);
+      #outside_cuts(length=110, finger=5, material=10, center=false);
+  }
+  
+  rotate([180, 0, 0])
+  translate([0, 0])
+  difference() {  
+    color("red")
+      square([110, 50]);
+      translate([55, 5])
+      #inside_cuts(length=110, finger=5, material=10, center=true);
+  }
+}
+
+//finger_testing();
+
+/*
+##module: outside_cuts
+Create a set of finger-joint cuts that result in two larger cuts taken at the outside
+edge
+  ###parameters:
+    *length* (real)         length of edge
+    *finger* (real)         length of each individual finger
+    *material* (real)       thickness of material - sets cut depth
+    *center* (boolean)         center the set of fingers with respect to origin
+*/
+
+module outside_cuts(length=6, finger=1, material=1, text=false, center=false) {
+  // overage to ensure that all cuts are completed
+  overage = 0.0001;
+
+
+  //maximum possible divisions for this length
+  max_divisions = floor(length/finger);
+  //number of usable divisions that fall completely within the edge
+  //for this implentation the number of divisions must be odd
+  usable_divisions = max_divisions%2==0 ? max_divisions-3 : max_divisions-2;
+
+  // number of "female cuts"
+  num_cuts = floor(usable_divisions/2);
+
+  //length of cut at either end
+  end_cut_length = (length-usable_divisions*finger)/2;
+
+  padding = end_cut_length + finger;
+
+  //set position relative to origin
+  x_translation = center==false ? 0 : -(num_cuts*2+1)*finger/2-end_cut_length;
+  y_translation = center==false ? 0 : -material/2;
+
+  translate([x_translation, y_translation]) {
+  
+    // make both endcuts here
+    for (j = [0, 1]) {
+      translate([(length-end_cut_length)*j, -overage/2])
+        square([end_cut_length, material+overage]);
+    }
+    
+
+    //make all the "normal" finger cuts here
+    for (i = [0 : num_cuts-1]) {
+      translate([i*finger*2+padding, -overage/2]) //move cuts slightly in y plane
+        square([finger, material+overage]);
+    
+    }
+  }
+
+} //end outside_cuts
+
+
+/*
+##module: inside_cuts
+Create a set of finger-joint cuts all of the same size  
+
+###parameters:
+    *length* (real)         length of edge
+    *finger* (real)         length of each individual finger
+    *material* (real)       thickness of material - sets cut depth
+    *center* (boolean)         center the set of fingers with respect to origin
+*/
+
+module inside_cuts(length=6, finger=1, material=1, text=false, center=false) {
```

Comment:
Looks like there's still the `text` parameter here and it's not being used. Same in `outside_cuts`

Reply:
In order to help with the principle of least surprise, could this module be tweaked such that when you place it un-centered at `[0, 0]` it meshes with results of `outside_cuts` with the same parameters? In my modifications, I just did a `difference()` between `outside_cuts` and the appropriate-length square.